src/fdb5/api/FDB.h

```diff
@@ -72,73 +76,200 @@ class FDB {
     FDB(FDB&&)            = default;
     FDB& operator=(FDB&&) = default;
 
-    // -------------- Primary API functions ----------------------------
+    // -------------- Primary API functions ----------------------------------------------------------------------------
 
+    /// Archive binary data to a FDB.
+    /// Any callback set with registerArchiveCallback will be invoked.
+    /// @param handle eckit::message::Message to data to archive
     void archive(eckit::message::Message msg);
+
+    /// Archive binary data to a FDB.
+    /// Reads messages from the eckit::DataHandle and calls archive() the the
+    /// corresponding messages.
+    /// Any callback set with registerArchiveCallback will be invoked on each message.
+    /// @param handle eckit::DataHandle reference data to archive
     void archive(eckit::DataHandle& handle);
+
+    /// Archive binary data to a FDB.
+    /// Internally creates a DataHandle and calls archive().
+    /// Any callback set with registerArchiveCallback will be invoked on each message.
+    /// @param data Pointer to the binary data to archive
+    /// @param length Size of the data to archive with the given
     void archive(const void* data, size_t length);
-    // warning: not high-perf API - makes sure that all the requested fields are archived and there are no data
-    // exceeding the request
+
+    /// Archives data from Datahandle and ensures all keys exactly match the provided MarsRequest.
+    /// Any callback set with registerArchiveCallback will be invoked on each message.
+    /// @param request a mars request
+    /// @param handle a data handle pointing to the data
+    /// @throws eckit::UserError if there are more keys in the MarsRequest then in the messages.
+    /// @throws eckit::UserError if message key not present in MarsRequest.
     void archive(const metkit::mars::MarsRequest& request, eckit::DataHandle& handle);
 
-    // disclaimer: this is a low-level API. The provided key and the corresponding data are not checked for consistency
-    // Optional callback function is called upon receiving field location from the store.
+    /// Archive binary data to a FDB.
+    /// Any callback set with registerArchiveCallback will be invoked.
+    /// @note No constistency checks are applied. The caller needs to ensure the provided key matches metadata present
+    /// in data.
+    /// @param key Key used for indexing and archiving the data
+    /// @param data Pointer to the binary data to archive
+    /// @param length Size of the data to archive with the given @p key
```

Comment:
Archives a single object.

Not sure I like the term 'binary data here' - we don't care what is in the blob. It is a blob. Call it an object, or similar.

Reply:
done